node/src/components/small_network.rs

```diff
@@ -710,8 +775,12 @@ where
                     self.update_peers_metric();
                     return remove;
                 }
+
+                // This speeds up the connection process, but masks potential bugs in the gossiper.
+                let effects = self.connect_to_peer_if_required(public_address);
                 self.update_peers_metric();
-                Effects::new()
+
+                effects
```

Comment:
If you call `update_peers_metric()` first, then you can return with `self.connect_to_peer`.

Reply:
Unfortunately, `connect_to_peer_if_required` manipulates `self.pending`, which is read by `update_peers_metric` via `self.peers`, so we'd be off by one by this change.

Reply:
I stand corrected, you're right!